lib/widgets/text.dart

```diff
@@ -1,5 +1,69 @@
+///
+/// Include this in the app-wide [MaterialApp.theme].
+///
+/// We expect these text styles to be the basis of all the styles chosen by the
+/// Material library's widgets, such as the default styling of
+/// an [AppBar]'s title, of an [ElevatedButton]'s label, and so on.
+///
+/// As of writing, it turns out that these styles flow naturally into
+/// most of our own widgets' text styles as well.
+/// We often see this in the child of a [Material], for example,
+/// since by default [Material] applies an [AnimatedDefaultTextStyle]
+/// with the [TextTheme.bodyMedium] that gets its value from here.
+///
+/// In all the comprised [TextStyle]s, this sets:
+///
+/// - [TextStyle.fontFamily] to [kDefaultFontFamily], and
+/// - [TextStyle.fontFamilyFallback] to [kDefaultFontFamilyFallback].
+///
+/// Since [kDefaultFontFamily] is a variable-weight font,
+/// each [TextStyle] needs some processing, which this provides,
+/// in order to correctly specify the font weight.
+///
+/// When building on top of these [TextStyles], callers that wish to specify
+/// a different font weight are still responsible for reprocessing the style
+/// with [weightVariableTextStyle] before passing it to a [Text].
+/// (Widgets in the Material library won't do this; they aren't yet equipped
+/// to set font weights on variable-weight fonts. If this causes visible bugs,
+/// we should investigate and fix, but they should become less likely as
+/// we transition from Material's widgets to our own bespoke ones.)
+// TODO decide if we like this data flow for our own widgets' text styles.
+//   Does our design fit well with the fields of a [TextTheme]?
+//   (That's [TextTheme.titleLarge], [TextTheme.bodyMedium], etc.)
+Typography zulipTypography(BuildContext context) {
+  final typography = Theme.of(context).typography;
+  return weightVariableTypography(context, typography)
+    .copyWith(
+      black: typography.black.apply(
+        fontFamily: kDefaultFontFamily,
+        fontFamilyFallback: kDefaultFontFamilyFallback),
+      white: typography.white.apply(
+        fontFamily: kDefaultFontFamily,
+        fontFamilyFallback: kDefaultFontFamilyFallback),
+    );
+}
+
+/// The [TextStyle.fontFamily] to use in most of the app.
+///
+/// The same [TextStyle] should also specify [kDefaultFontFamilyFallback]
+/// for [TextStyle.fontFamilyFallback].
+///
+/// This is a variable-weight font, so any [TextStyle] that uses this should be
+/// merged with the result of calling [weightVariableTextStyle].
+const kDefaultFontFamily = 'Source Sans 3';
+
+/// The [TextStyle.fontFamilyFallback] for use with [kDefaultFontFamily].
+final kDefaultFontFamilyFallback = [
+  // iOS doesn't support any of the formats this font is available in.
+  // If we use it on iOS, we'll get blank spaces where we could have had Apple-
+  // style emojis.
+  if (defaultTargetPlatform == TargetPlatform.android) 'Noto Color Emoji',
```

Comment:
This is a bit of a perilous pattern because `defaultTargetPlatform` can change over time. Not likely to happen in a running app — but it happens all the time in tests, which indeed is the point of `defaultTargetPlatform`.

So as a static final, `kDefaultFontFamilyFallback` will get computed once when it's first accessed, and subsequent accesses will get the same value even if `defaultTargetPlatform` has changed. If we have a test that cares about the value, then the test could pass or fail depending on which other tests ran first. That in turn can be a big pain for debugging why some test is behaving the way it is (e.g. for figuring out why some other change broke a test).

I think a good fix would be to make it a getter, just like `defaultTargetPlatform` itself:
```dart
List<String> get defaultFontFamilyFallback => [
```

There's no performance need to memoize it, because at the tip of the PR we're only calling this twice per build of `ZulipApp`.